### ISSUE TITLE
feat(VariantManagement): Add features to comply with UX Guidelines

### ISSUE THREAD

**Issue by Lukas742** [SAP/ui5-webcomponents-react#1609]
Currently the `VariantManagement` only implements the basic behavior. We should enhance the component to comply with the UX Guidelines.

- [Design Guidelines](https://experience.sap.com/fiori-design-web/variant-management/)
- [UI5 API](https://sapui5.hana.ondemand.com/1.90.0/#/api/sap.ui.comp.variants.VariantManagement)

**Comment by kartikc13** [comment]
@MarcusNotheis : I have following clarifications needed on this topic.
1. Is the permanent persistence of the variants also included in your design or does it just caters to the UI control handling?
2. Is there a plan to persist and manage the variants via some central service or each application (MMP) is expected to build their own persistence? If it's the latter case then is there any guidance on the design we should follow?
3. Also if it is possible, can you share the design details on how the control will handle the variant configuration (JSON schema, etc.) and how data has to be bound to it?

**Comment by MarcusNotheis** [comment]
> 1. Is the permanent persistence of the variants also included in your design or does it just caters to the UI control handling?

No, this is not part of the scope of this project. We are only providing the rendering (= UI Layer) of this component. All other aspects like data-binding and persistence is up to the end users.
 
> 2. Is there a plan to persist and manage the variants via some central service or each application (MMP) is expected to build their own persistence? If it's the latter case then is there any guidance on the design we should follow?

Same answer as for question 1: we don't provide any central services, so you would have to come up with your own persistence. Maybe you could reuse the flexibility service from the Fiori Launchpad if you transform the data in the correct format (but I unfortunately don't have any docs on this topic).

> 3. Also if it is possible, can you share the design details on how the control will handle the variant configuration (JSON schema, etc.) and how data has to be bound to it?

Currently our API looks like:
```jsx
<VariantManagement
  selectedKey="2"
  variantItems={[
    {
      key: '1',
      label: 'Variant 1'
    },
    {
      key: '2',
      label: 'Variant 2'
    }
  ]}
/>
```

I also could imagine that we rework the API to match other APIs in our project like:
```jsx
<VariantManagement>
  <VariantManagementItem id="1">Variant 1</VariantManagementItem>
  <VariantManagementItem id="2" selected>Variant 2</VariantManagementItem>
</VariantManagement>
```
Does this answer your question? Or did I miss something?

**Comment by kartikc13** [comment]
@MarcusNotheis : Thanks for the clarifications about Q1 and 2. Regarding Q3, I am aware of the current API instead what I am interested is what will be the API after the next release i.e. once we have included the features like add, remove and apply variant. For example, once we add a new variant in what JSON structure would the configurations, of lets say FilterBar, would be captured?

**Comment by d-schroeder** [comment]
What's the status on this? In order to save new smart variants we need a save button on the Variant Manager. Any info when this may be implemented?

**Comment by Lukas742** [comment]
Hi @d-schroeder 

we've planned to release this by the end of next week. If something comes up that interferes with that plan, I'll let you know.

**Comment by Lukas742** [comment]
Hi @d-schroeder 

sorry, but our ETA for the VariantManagement has changed to next week. Unfortunately we had some urgent task that did not allow any postponement. Also when looking deeper into the implementation of the UI5 `VariantManagement` we noticed, that this is not as simple as expected. By the end of the day I will open a Draft PR so you can track the progress of our implementation there.
